<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GraphicsEditor</class>
 <widget class="QMainWindow" name="GraphicsEditor">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1096</width>
    <height>863</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>黑体</family>
    <weight>50</weight>
    <bold>false</bold>
   </font>
  </property>
  <property name="windowTitle">
   <string>GraphicsEditor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="GraphicsEditor.qrc">
    <normaloff>:/GraphicsEditor/windowIcon</normaloff>:/GraphicsEditor/windowIcon</iconset>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonTextUnderIcon</enum>
  </property>
  <property name="animated">
   <bool>true</bool>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="dockNestingEnabled">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="PaintArea" name="paintArea" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::DefaultContextMenu</enum>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="openAction"/>
   <addaction name="saveAction"/>
   <addaction name="separator"/>
   <addaction name="eraseAction"/>
   <addaction name="separator"/>
   <addaction name="lineAction"/>
   <addaction name="polygonAction"/>
   <addaction name="circleAction"/>
   <addaction name="ellipseAction"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="openAction">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="GraphicsEditor.qrc">
     <normaloff>:/GraphicsEditor/open</normaloff>:/GraphicsEditor/open</iconset>
   </property>
   <property name="text">
    <string>open</string>
   </property>
   <property name="statusTip">
    <string>打开</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="saveAction">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="GraphicsEditor.qrc">
     <normaloff>:/GraphicsEditor/save</normaloff>:/GraphicsEditor/save</iconset>
   </property>
   <property name="text">
    <string>save</string>
   </property>
   <property name="statusTip">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="lineAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="GraphicsEditor.qrc">
     <normaloff>:/GraphicsEditor/line</normaloff>:/GraphicsEditor/line</iconset>
   </property>
   <property name="text">
    <string>line</string>
   </property>
   <property name="statusTip">
    <string>直线</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::WindowShortcut</enum>
   </property>
   <property name="autoRepeat">
    <bool>true</bool>
   </property>
   <property name="visible">
    <bool>true</bool>
   </property>
   <property name="shortcutVisibleInContextMenu">
    <bool>false</bool>
   </property>
  </action>
  <action name="polygonAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="GraphicsEditor.qrc">
     <normaloff>:/GraphicsEditor/polygon</normaloff>:/GraphicsEditor/polygon</iconset>
   </property>
   <property name="text">
    <string>polygon</string>
   </property>
   <property name="statusTip">
    <string>多边形</string>
   </property>
  </action>
  <action name="circleAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="GraphicsEditor.qrc">
     <normaloff>:/GraphicsEditor/round</normaloff>:/GraphicsEditor/round</iconset>
   </property>
   <property name="text">
    <string>circle</string>
   </property>
   <property name="statusTip">
    <string>圆</string>
   </property>
  </action>
  <action name="ellipseAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="GraphicsEditor.qrc">
     <normaloff>:/GraphicsEditor/ellipse</normaloff>:/GraphicsEditor/ellipse</iconset>
   </property>
   <property name="text">
    <string>ellipse</string>
   </property>
   <property name="statusTip">
    <string>椭圆</string>
   </property>
  </action>
  <action name="eraseAction">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="GraphicsEditor.qrc">
     <normaloff>:/GraphicsEditor/erase</normaloff>:/GraphicsEditor/erase</iconset>
   </property>
   <property name="text">
    <string>erase</string>
   </property>
   <property name="toolTip">
    <string>橡皮擦</string>
   </property>
   <property name="statusTip">
    <string>擦除</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PaintArea</class>
   <extends>QWidget</extends>
   <header location="global">paintarea.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="GraphicsEditor.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>lineAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>GraphicsEditor</receiver>
   <slot>selectLineShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>547</x>
     <y>431</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ellipseAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>GraphicsEditor</receiver>
   <slot>selectEllipseShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>362</x>
     <y>66</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>polygonAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>GraphicsEditor</receiver>
   <slot>selectPolygonShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>329</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>circleAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>GraphicsEditor</receiver>
   <slot>selectRoundShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>281</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>eraseAction</sender>
   <signal>toggled(bool)</signal>
   <receiver>GraphicsEditor</receiver>
   <slot>selectEraseShape()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>213</x>
     <y>67</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>paintArea</sender>
   <signal>windowTitleChanged(QString)</signal>
   <receiver>GraphicsEditor</receiver>
   <slot>setWinTitle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>574</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>592</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>selectLineShape()</slot>
  <slot>selectEllipseShape()</slot>
  <slot>selectRoundShape()</slot>
  <slot>selectPolygonShape()</slot>
  <slot>selectEraseShape()</slot>
  <slot>newImage()</slot>
  <slot>setWinTitle()</slot>
 </slots>
</ui>
